src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5196,7 +5218,7 @@ private VariableState GetVariableState(Optional<LocalState> localState)
             return new VariableState(
                 _variableSlot.ToImmutableDictionary(),
                 ImmutableArray.Create(variableBySlot, start: 0, length: nextVariableSlot),
-                _variableTypes.ToImmutableDictionary(),
+                _variableTypes.ToImmutableDictionary(_variableTypes.Comparer, TypeWithAnnotations.EqualsComparer.ConsiderEverythingComparer),
```

Comment:
>_variableTypes.Comparer, TypeWithAnnotations.EqualsComparer.ConsiderEverythingComparer [](start = 53, length = 86)

Is this change needed? (The key comparer is the default, and the value comparer doesn't appear to be used.)

Reply:
Yes, without this change, the bootstrap build crashes because it attempts to call `TypeWithAnnotations.Equals` on the dictionary values.

Reply:
Where are we calling `Equals` on the dictionary values?

---
In reply to: [451162769](https://github.com/dotnet/roslyn/pull/45527#discussion_r451162769) [](ancestors = 451162769)

Reply:
`ToImmutableDictionary()` may call Equals on dictionary values internally when constructing the immutable dictionary. This is something we might want to examine in greater detail. It's not obvious to me why sharing the `_variableTypes` between a lambda and its containing method would make this behavior start to surface in the bootstrap build.